clippy_lints/src/empty_with_brackets.rs

```diff
@@ -97,22 +115,94 @@ impl EarlyLintPass for EmptyWithBrackets {
         }
     }
 
-    fn check_variant(&mut self, cx: &EarlyContext<'_>, variant: &Variant) {
+    fn check_variant(&mut self, cx: &LateContext<'_>, variant: &Variant<'_>) {
+        // Don't lint pub enums
+        if cx.effective_visibilities.is_reachable(variant.def_id) {
```

Comment:
Aren't `pub enum` variants still ok to lint if they are struct variants, given that they don't have a constructor? E.g. for `pub enum X { V {} }`,  `X::V` doesn't have a constructor that could be used as a value.

(Comment is based on the lintcheck result: https://github.com/rust-lang/rust-clippy/actions/runs/10386147840)

Or is this visibility check unrelated to tuple variant constructs being used as values and you just don't want the lint to emit for public items in general?